crates/egui-wgpu/src/lib.rs

```diff
@@ -224,6 +294,40 @@ pub fn depth_format_from_bits(depth_buffer: u8, stencil_buffer: u8) -> Option<wg
 
 // ---------------------------------------------------------------------------
 
+/// A human-readable summary about an adapter
+pub fn adapter_info_summary(info: &wgpu::AdapterInfo) -> String {
+    let wgpu::AdapterInfo {
+        name,
+        vendor: _, // skip integer id
+        device: _, // skip integer id
```

Comment:
vendor and device id as hex can be super useful though! For instance wgpu uses that internally to enable workarounds.
Bonus points if you use this (pub!!) table https://github.com/gfx-rs/wgpu/blob/trunk/wgpu-hal/src/auxil/mod.rs#L7 to attempt stringification!